service/history/historyEngine.go

```diff
@@ -1593,6 +1582,208 @@ func (e *historyEngineImpl) SignalWorkflowExecution(signalRequest *h.SignalWorkf
 		})
 }
 
+func (e *historyEngineImpl) SignalWithStartWorkflowExecution(signalWithStartRequest *h.SignalWithStartWorkflowExecutionRequest) (
+	*workflow.StartWorkflowExecutionResponse, error) {
+
+	domainID, err := getDomainUUID(signalWithStartRequest.DomainUUID)
+	if err != nil {
+		return nil, err
+	}
+	sRequest := signalWithStartRequest.SignalWithStartRequest
+	execution := workflow.WorkflowExecution{
+		WorkflowId: sRequest.WorkflowExecution.WorkflowId,
+		RunId:      sRequest.WorkflowExecution.RunId,
+	}
+
+	isBrandNew := false
+	prevRunID := ""
+	attempt := 0
+
+	context, release, err0 := e.historyCache.getOrCreateWorkflowExecution(domainID, execution)
+	if err0 != nil {
+		if _, ok := err0.(*workflow.EntityNotExistsError); ok {
+			isBrandNew = true
+			goto Start_And_Signal
```

Comment:
why do we use goto?

Reply:
Changed.

Reply:
agree it might feel uncomfortable, removed.